File: APP MOBILE/smart_money/lib/modules/wallet/models/transfer_request.dart
/// Request chuyển tiền giữa các ví.
/// Tương ứng: TransferRequest.java (server)
class TransferRequest {
  final int fromWalletId;
  final int toWalletId;
  final double amount;

  const TransferRequest({
    required this.fromWalletId,
    required this.toWalletId,
    required this.amount,
  });

  Map<String, dynamic> toJson() {
    return {
      'fromWalletId': fromWalletId,
      'toWalletId': toWalletId,
      'amount': amount,
    };
  }
}

File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/wallet/TransferRequest.java
package fpt.aptech.server.dto.wallet;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
public class TransferRequest {
    @NotNull(message = "Source wallet ID is required")
    private Integer fromWalletId;

    @NotNull(message = "Destination wallet ID is required")
    private Integer toWalletId;

    @NotNull(message = "Transfer amount is required")
    @Positive(message = "Transfer amount must be greater than 0")
    private BigDecimal amount;

    private String note;
}
